### PR TITLE
#14641. Reset committer on dbtable destruction to avoid use-after-free

### PR DESCRIPTION
**Pull request by cb-mega**

### CODE REVIEW

File: include/mega/db.h [hunk 1]
```diff
@@ -35,9 +35,11 @@ class MEGA_API DbTable
 
 protected:
     bool mCheckAlwaysTransacted = false;
-    DBTableTransactionCommitter* mCurrentTransactionCommiter = nullptr;
+    DBTableTransactionCommitter* mTransactionCommitter = nullptr;
```
Comment (cb-mega) [inline include/mega/db.h]: Better naming

File: include/mega/db.h [hunk 2]
```diff
@@ -88,7 +90,7 @@ class MEGA_API DBTableTransactionCommitter
     bool mStarted = false;
 
 public:
-    inline void beginOnce()
+    void beginOnce()
```
Comment (cb-mega) [inline include/mega/db.h]: the methods are already inline. No need for superfluous `inline` keyword.

File: include/mega/db.h [hunk 3]
```diff
@@ -109,30 +111,37 @@ class MEGA_API DBTableTransactionCommitter
...
             }
             else
             {
-                mTable->mCurrentTransactionCommiter = this;
+                mTable->mTransactionCommitter = this;
             }
         }
     }
+
+    MEGA_DISABLE_COPY_MOVE(DBTableTransactionCommitter)
```
Comment (cb-mega) [inline include/mega/db.h]: Without this the committer is copyable but accidentally copying it would be UB.

File: src/db.cpp [hunk 4]
```diff
@@ -84,17 +84,28 @@ void DbTable::checkTransaction()
...
+            mTransactionCommitter->beginOnce();
         }
     }
 }
 
+void DbTable::resetCommitter()
+{
+    if (mTransactionCommitter)
+    {
+        mTransactionCommitter->commitNow();
```
Comment (sergiohs84) [inline src/db.cpp]: We may not want to commit the open transaction in all cases. Remember the `sctable` should only commit upon reception of a new `scsn` via actionpackets (or a bit later when the ongoing cs requests complete).
Please, check that it's not possible to do a commit by mistake, or the account's state may get corrupted. Thanks

Comment (cb-mega) [reply src/db.cpp]: Yeah, I think you're right. In fact, I think we shouldn't commit here at all as this function is only called from the table destructor. Anything that wasn't committed until then should be discarded (I assume that's the reason why we're calling `abort` in the table destructor?!).

Comment (cb-mega) [reply src/db.cpp]: Ok. Done. And now the method name makes more sense :)

**Pull request merged**